plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -273,13 +280,52 @@ public void onFailure(Exception e) {
+             * the request context with the new values.
+             *
+             * @param incomeQueryObject The object representing the income query to be updated.
+             * @param outputMapping A map containing the mapping between new query fields and model output field names.
+             * @param mlOutput The MLOutput object containing the results from the machine learning model.
+             * @param requestContext The context object for the current pipeline processing request.
+             *
+             * @throws IllegalArgumentException If a specified JSON path cannot be found in the query string.
+             *
+             * @implNote This method uses JsonPath for JSON manipulation and supports both regular and extended (ext) fields.
+             *           For extended fields, it creates nested structures if they don't exist.
+             *           The method also updates the request context with new field values for further processing.
+             *
+             * @see JsonPath
+             * @see PipelineProcessingContext
+             * @see MLOutput
+             */
+            private void updateIncomeQueryObject(
+                Object incomeQueryObject,
+                Map<String, String> outputMapping,
+                MLOutput mlOutput,
+                PipelineProcessingContext requestContext
+            ) {
                 for (Map.Entry<String, String> outputMapEntry : outputMapping.entrySet()) {
-                    String newQueryField = outputMapEntry.getKey();
-                    String modelOutputFieldName = outputMapEntry.getValue();
-                    Object modelOutputValue = getModelOutputValue(mlOutput, modelOutputFieldName, ignoreMissing, fullResponsePath);
-                    String jsonPathExpression = "$." + newQueryField;
-                    JsonPath.parse(incomeQueryObject).set(jsonPathExpression, modelOutputValue);
+                    String newQueryField = null;
+                    try {
+                        newQueryField = outputMapEntry.getKey();
+                        String modelOutputFieldName = outputMapEntry.getValue();
+                        Object modelOutputValue = getModelOutputValue(mlOutput, modelOutputFieldName, ignoreMissing, fullResponsePath);
+
+                        if (newQueryField.startsWith("$.ext.") || newQueryField.startsWith("ext.")) {
+                            incomeQueryObject = StringUtils.prepareNestedStructures(incomeQueryObject, newQueryField);
+                        }
+
+                        JsonPath.using(suppressExceptionConfiguration).parse(incomeQueryObject).set(newQueryField, modelOutputValue);
+
+                        requestContext.setAttribute(newQueryField, modelOutputValue);
+
+                    } catch (PathNotFoundException e) {
+                        throw new IllegalArgumentException("can not find path " + newQueryField + "in query string");
```

Comment:
do you want to log the full exception message here?

Reply:
added logs